### PR TITLE
Update templates to handle missing node BGP IP

### PR DESCRIPTION
**Pull request by robbrockbank**
Check that the IPv4 address key is present before trying to get the value.

It's a very "mainline" change in the template so it's getting good coverage in the STs, but there is no ST to test the actual issue that led to us wanting this fix (since that would require jimmying around in etcd).

### CODE REVIEW

**Comment by insequent** [comment]
Hey @robbrockbank, I'm seeing issues similar to this, when trying to use calico/routereflector. My issue may be different than your intent with this PR, but the timing of your change was impeccable. I'm curious if maybe you saw something similar to what I was seeing?

I noticed the routereflector's bird.cfg.template failing in my confd log:
```
2017-03-16T00:56:22Z mgmt-01 /confd[15]: DEBUG Compiling source template /templates/bird.cfg.template
2017-03-16T00:56:22Z mgmt-01 /confd[15]: ERROR template: bird.cfg.template:66:71: executing "bird.cfg.template" at <getv $cnode_ip_key>: errorcalling getv: key does not exist
```
When I started checking the keys, I noticed a discrepancy in the hostnames:
```
"key": "/calico/bgp/v1/host/control-02.conant01.com",
  "key": "/calico/bgp/v1/host/control-02.conant01.com/peer_v4",
    "key": "/calico/bgp/v1/host/control-02.conant01.com/peer_v4/10.232.255.4",
    "value": "{\"ip\":\"10.232.255.4\",\"as_num\":\"64512\"}"
  "key": "/calico/bgp/v1/host/control-02",
    "key": "/calico/bgp/v1/host/control-02/ip_addr_v4",
    "value": "10.232.255.6"
    "key": "/calico/bgp/v1/host/control-02/ip_addr_v6",
    "value": ""
```
So in other words, the template is failing because while calico was generating these etcd entries, something switched it between using short and long hostnames. I'm trying to narrow down where this is happening, and I'm hopeful you might have seen this somewhere on your side. If not, please ignore this rudely long PR comment!

**Comment by insequent** [comment]
Found the issue. It was a misconfigured ansible task. Deleting these comments to remove the spam.
